### DISCUSSION TITLE
Getters in nested objects

### DISCUSSION THREAD

**Discussion by Mosquid** [typegoose/typegoose#715] [Q&A]
Hey. How do I make getters work for an array of nested objects?

```ts
class ChildClass {
  @prop()
  public description: string;
  
  public get shortDescription () {
    return trim(this.description, 50)
  }
}

class ParentClass {
   @prop({ getters: true,  type: () => [ChildClass] })
   public children: ChildClass[];
}
```

**Comment by smolinari** [comment]
@Mosquid - What isn't working? Are you getting an error? 

Scott

**Reply by Mosquid** [reply]
No, it just doesn't seem to call "toObject" against the "children" property. `parentClass.children[0].shortDescription === undefined`

**Reply by hasezoey** [reply]
> No, it just doesn't seem to call "toObject" against the "children" property.

This sounds like [Why is my virtual not included in the output?](https://typegoose.github.io/typegoose/docs/api/virtuals/#why-is-my-virtual-not-included-in-the-output).

**Reply by Mosquid** [reply]
> This sounds like [Why is my virtual not included in the output?](https://typegoose.github.io/typegoose/docs/api/virtuals/#why-is-my-virtual-not-included-in-the-output).

I have this my schemaOptions and virtuals are included as for the "ParentClass":

```
{...
   toJSON: { virtuals: true },
   toObject: { virtuals: true }
}
```

Although, there's a slightly different case shown by the link. I'm not using Ref

**Reply by hasezoey** [reply]
then maybe try adding `getters: true`, like `virtuals: true` (getters are not virtuals) ([see this](https://mongoosejs.com/docs/tutorials/getters-setters.html))

PS: maybe also try adding this to the child's schema options, because i dont know if it will populate down

**Reply by Mosquid** [reply]
Please pardon me! I missed `modelOptions` for my child model. Thank you
